packages/teams-js/src/public/app.ts

```diff
@@ -565,6 +565,26 @@ export namespace app {
    */
   const initializationTimeoutInMs = 5000;
 
+  function logWhereTeamsJsIsBeingUsed(): void {
+    if (inServerSideRenderingEnvironment()) {
+      return;
+    }
+    const scripts = document.getElementsByTagName('script');
+    // This will always be the current script because browsers load and execute scripts in order.
+    // Whenever a script is executing for the first time it will be the last script in this array.
+    const currentScriptSrc = scripts && scripts[scripts.length - 1] && scripts[scripts.length - 1].src;
+    const scriptUsageWarning =
+      'Today, teamsjs can only be used from a single script or you may see undefined behavior. If you would like to use teamjs from more than one script at the same time, please open an issue at https://github.com/OfficeDev/microsoft-teams-library-js/issues';
```

Comment:
>Today, teamsjs can only be used from a single script or you may see undefined behavior. [](http://example.com/codeflow?start=7&length=87)

Given that this line will be logged unconditionally (and not just when it detects a multi-usage scenario), it might be slightly clearer if it was reworded to indicate that? E.g.,

"Today TJS can only be used.... This log line is used to help detect cases where TJS is loaded multiple times -- it is always written. The presence of the log itself does not indicate a multi-load situation, but multiples of these log lines will."

Or something like that?